source/base/exceptions.cc

```diff
@@ -336,6 +343,45 @@ void ExceptionBase::generate_message () const
 
 
 
+ExcMPI::ExcMPI (const int error_code)
+  :
+  error_code (error_code)
+{}
+
+void ExcMPI::print_info (std::ostream &out) const
+{
+#ifdef DEAL_II_WITH_MPI
+  char error_name[MPI_MAX_ERROR_STRING];
+  error_name[0] = '\0';
+  int resulting_length = MPI_MAX_ERROR_STRING;
+  // get the string name of the error code by first converting it to an error
+  // class.
+  int error_class = 0;
+  // there is nothing we can do if these functions fail (they just provide
+  // better input), so assume that the given error code is valid.
```

Comment:
The output is the message printed to the screen. I assumed that if we get this far then error code is a real MPI error code and the message printing functions will understand it.

Now that I look at things again I think it would be worthwhile to check these return codes and print a shorter message if one of them fails (which is unlikely to happen but the check is easy).